### PR TITLE
Adding #2982 and #3074#issuecomment-591726341

### PR DESCRIPTION
**Pull request by ShrutiRawal**
Fix for Issue #2982  and alert as requested by https://github.com/ccrama/Slide/issues/3074#issuecomment-591726341

### CODE REVIEW

**Comment by TacoTheDank** [comment]
@ShrutiRawal Hi, I hope you don't mind if I make some small edits to your PR. I figure it'd be faster than manually pointing out every little nitpick that I have :)

**Comment by ShrutiRawal** [comment]
Yes sure @TacoTheDank , you can proceed...

**Comment by TacoTheDank** [comment]
I've merged this into another branch for now to be able to work with it easier. Did you build an APK and test the changes?

**Comment by TacoTheDank** [comment]
Since there's too many technicalities with the current implementation of the settings, I will add these changes to the settings rewrite branch.

File: app/src/main/java/me/ccrama/redditslide/Activities/MainActivity.java [hunk 1]
```diff
@@ -828,16 +837,17 @@ public void onClick(@NonNull MaterialDialog materialDialog,
...
         inNightMode = SettingValues.isNight();
         disableSwipeBackLayout();
-        super.onCreate(savedInstanceState);
-        if ((getIntent().getFlags() & Intent.FLAG_ACTIVITY_BROUGHT_TO_FRONT) != 0) {
-            // Activity was brought to front and not created
-            finish();
-            return;
-        }
-        if (!Slide.hasStarted) {
-            Slide.hasStarted = true;
```
Comment (TacoTheDank) [inline app/src/main/java/me/ccrama/redditslide/Activities/MainActivity.java:840]: Can you explain your changes here? Why did you move the super call up? Why was this stuff removed?

File: app/src/main/res/layout/activity_media.xml [hunk 2]
```diff
@@ -97,6 +97,7 @@
             android:layout_height="56dp"
             android:layout_toLeftOf="@+id/more"
             android:theme="@style/Ripple.List"
+            android:onClick="saveMedia"
```
Comment (TacoTheDank) [inline app/src/main/res/layout/activity_media.xml:100]: The save button already had an OnClickListener, and you also didn't account for the bottomsheet save ability.

File: app/src/main/res/layout/activity_settings_datasaving_child.xml [hunk 3]
```diff
@@ -103,6 +103,56 @@
...
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:text="@string/btn_submit"
+                    android:onClick="setDataLimit"
+                    android:id="@+id/btn_submitDataLimit" />
+
+            </LinearLayout>
+
+        </RelativeLayout>
+
```
Comment (TacoTheDank) [inline app/src/main/res/layout/activity_settings_datasaving_child.xml:155]: This should've been added after the following switch button, as that switch is part of the second category.

File: app/src/main/java/me/ccrama/redditslide/Activities/MediaView.java [hunk 4]
```diff
@@ -1226,4 +1229,23 @@ public void onFolderSelection(@NonNull FolderChooserDialogCreate dialog,
...
+        if (txtMediaSize.getText()!=null && txtMediaSize.getText().toString().length()!=0){
+            String size = txtMediaSize.getText().toString();
+            size=size.substring(0,size.length()-1);  // to remove 'MB'
+            SharedPreferences preferences = getSharedPreferences("prefs", Context.MODE_PRIVATE);
+            String alertLimit = preferences.getString("limit","1000");  //we can set 1000 MB to be default data alert limit
+            if(Integer.parseInt(size)>Integer.parseInt((alertLimit))){
+                Toast.makeText(MediaView.this,R.string.err_media_greaterThanLimit,Toast.LENGTH_SHORT).show();
+            }
+        }
+    }
```
Comment (TacoTheDank) [inline app/src/main/java/me/ccrama/redditslide/Activities/MediaView.java:1250]: Also this doesn't at all close #3074. The issue creator's request was to use a HEAD request to show the size of the media on the post cards. This implementation only deals with when the save button is clicked, which is effectively redundant as the post is still being loaded regardless (because it's a separate task). The actual post loading is what concerns the issue creator, not saving the media.

File: app/src/main/java/me/ccrama/redditslide/Activities/SettingsData.java [hunk 5]
```diff
@@ -24,6 +39,35 @@ public void onCreate(Bundle savedInstanceState) {
...
+    }
+
+    //returns the current data limit set
+    public int getDataLimit(){
+        String dlimit = preferences.getString("limit","");
+        if(dlimit.length()==0){
+            return -1;
+        }else{
+            return Integer.parseInt(dlimit);
+        }
```
Comment (TacoTheDank) [inline app/src/main/java/me/ccrama/redditslide/Activities/SettingsData.java:70]: This should've been added to the SettingsDataFragment class.

File: app/src/main/java/me/ccrama/redditslide/Activities/MediaView.java (hunk 4 above)
Comment (ShrutiRawal) [reply app/src/main/java/me/ccrama/redditslide/Activities/MediaView.java:1250]: Okayy, I'll work on this and make another PR using the mentioned approach in sometime :)

File: app/src/main/java/me/ccrama/redditslide/Activities/MainActivity.java (hunk 1 above)
Comment (ShrutiRawal) [reply app/src/main/java/me/ccrama/redditslide/Activities/MainActivity.java:840]: I was trying to figure out it's use, so tried to remove and observe. I was supposed to restore it before committing the changes....Looks like I forgot to do that...Sorry!!
